packages/terra-navigation-layout/src/configurationPropTypes.js

```diff
@@ -0,0 +1,107 @@
+import PropTypes from 'prop-types';
+
+const supportedComponentBreakpoints = ['tiny', 'small', 'medium', 'large', 'huge'];
+
+/**
+ * PropType definition for component definitions in the NavigationLayout's configuration object.
+ * It is an Object that is keyed by a String breakpoint value (or `default`) and has Object values that
+ * represent Components.
+ *
+ * Example:
+ *   {
+ *     default: {
+ *       componentClass: DefaultComponent,
+ *       props: {
+ *         prop1: 'Value',
+ *         prop2: 'Other value',
+ *       },
+ *     },
+ *     small: {
+ *       componentClass: SmallComponent,
+ *       // props are optional
+ *     },
+ *     huge: {
+ *       // The same component can be rendered with different prop values at different breakpoints
+ *       componentClass: DefaultComponent,
+ *       props: {
+ *         prop1: 'Huge Value for prop1',
+ *       },
+ *     },
+ *   }
+ */
+const componentConfigPropType = PropTypes.objectOf((propValue, key, componentName, location, propFullName) => {
+  const keyIsValid = key === 'default' || supportedComponentBreakpoints.indexOf(key) >= 0;
```

Comment:
Nitpick:

validKey and validInput feel like variable names that would make this code easier to read IMO.

Reply:
I agree, I will update this.